sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -588,6 +588,13 @@ object LimitPushDown extends Rule[LogicalPlan] {
     }
   }
 
+  private def isSupportPushdownThroughWindow(
+      windowExpressions: Seq[NamedExpression]): Boolean = windowExpressions.forall {
+    case Alias(WindowExpression(_: RankLike | _: RowNumberLike,
+        WindowSpecDefinition(Nil, _, _)), _) => true
```

Comment:
Does it really work with any kind of window frames? Can we add some comments to explain it?

Reply:
The window frame of RankLike and RowNumberLike is UNBOUNDED PRECEDING to CURRENT ROW.
https://github.com/apache/spark/blob/32a523b56fc329e858f3a463314c13e3710be5e0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala#L514